WakaTime.py

```diff
@@ -279,21 +327,41 @@ def enough_time_passed(now, is_write):
             or (now - _last_hb['timestamp'] > (2 if is_write else HEARTBEAT_FREQUENCY * 60)))
 
 
+@lru_cache(maxsize=4)
```

Comment:
I would leave it at default 128, as its return value is a short string. Not much of a memory footprint.
```suggestion
@lru_cache
```

Reply:
Weird, removing the `maxsize` completely, python throws an error.
```
lru_cache
    raise TypeError('Expected maxsize to be an integer or None')
TypeError: Expected maxsize to be an integer or None
```
So I changed the value to 128.